### PR TITLE
Allow FIFO pool in redis client

### PR DESCRIPTION
**Pull request by ktaekwon000**
Linked issue: #1819 

Allows the user to choose between fifo and lifo pool via passing in the PoolType variable into redis.Options.

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -76,6 +76,10 @@ type Options struct {
 	// Default is ReadTimeout.
 	WriteTimeout time.Duration
 
+	// Type of Pool
+	// Default is lifo, supports fifo too
+	// Note that fifo has higher overhead compared to lifo
+	PoolType pool.PoolType
```
Comment (vmihailenco) [inline options.go:82]: `pool` is a private package. Perhaps `PoolFIFO bool`?

Comment (ktaekwon000) [reply options.go:82]: Thanks for the headsup, I'll do that .

Comment (ktaekwon000) [reply options.go]: I've changed it, along with the golint issues.

File: internal/pool/pool.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/go-redis/redis/v8/internal"
 )
 
+type PoolFIFO bool
```
Comment (monkey92t) [inline internal/pool/pool.go:14]: It seems redundant

**Comment by monkey92t** [comment]
Other clients, we should also add it, for example: cluster, ring, sentinel, universal...

**Pull request merged**